src/TrackGenerator.cpp

```diff
@@ -1047,6 +1049,61 @@ void TrackGenerator::initializeTrackUids() {
 
 
 /**
+ * @brief Computes the volumes/areas of each Cell and Material in the Geometry.
+ * @details This computes the volumes/areas of each Cell and Material in the 
+ *          Geometry from the length of track segments crossing each Cell. This
+ *          routine assigns the volume and number of instances to each Cell and
+ *          Material. This is a helper routine that is called after track 
+ *          segmentation is complete in TrackGenerator::generateTracks().
+ */
+void TrackGenerator::initializeVolumes() {
+
+  if (!containsTracks())
+    log_printf(ERROR, "Unable to initialize volumes since tracks "
+               "have not yet been generated");
+
+  Cell* cell;
+  Material* material;
+  int num_FSRs = _geometry->getNumFSRs();
+  FP_PRECISION* fsr_volumes = getFSRVolumes();
+
+  /* Compute volume and number of instances for each Cell and Material */
+  for (int i=0; i < num_FSRs; i++) {
+    cell = _geometry->findCellContainingFSR(i);
+    cell->setVolume(cell->getVolume() + fsr_volumes[i]);
+    cell->setNumInstances(cell->getNumInstances() + 1);
```

Comment:
Currently, the only use of `Cell::setNumInstances()` and `Material::setNumInstances()` is to increment the number of instances. Could this just be `incrementNumInstances()` and there be no method argument?